### PR TITLE
Madeleine interactive statusbar

### PR DESCRIPTION
**Pull request by GeOdin**
The status bar is now interactive.

## Requirements:
* merge madeleine-add-tabwidget into master
* Pull request: https://github.com/nens/lizard_qgis_plugin/pull/8

### CODE REVIEW

**Comment by GeOdin** [comment]
[QGIS plug-in lizard (DockWidget) algemeen design [3]](https://trello.com/c/3arYe1K3/1304-qgis-plug-in-lizard-dockwidget-algemeen-design-3)

File: utils/set_dockwidget_gui.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# -*- coding: utf-8 -*-
+"""Module to change the dockwidget dynamically."""
+
+
+def status_bar_text(self, string):
```
Comment (jackieleng) [inline utils/set_dockwidget_gui.py]: Status bar variable names are not descriptive (`status_bar_2, status_bar_3... `)

**Review by jackieleng — CHANGES_REQUESTED** [review]

File: utils/set_dockwidget_gui.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# -*- coding: utf-8 -*-
```
Comment (jackieleng) [inline utils/set_dockwidget_gui.py]: I think this whole logic should be moved to ``dockwidget.py``, and into the dockwidget class itself. Basically, you're operating on the dockwidget, and if that's the case, than we probably should group that logic together.

Additional points:
- you're repeating the same things three times, better to use a loop
- I think a more elegant solution is to do something like this (more reusable also):

```python
def set_status_bar_text(status_bar, text):
    status_bar.setText(text)
    status_bar.repaint()
```

- ``status_bar_text`` as a function name is unclear to me what it does. Only after reading the code would you understand that it sets all three status bars.

File: lizard_downloader.py [hunk 3]
```diff
@@ -209,12 +204,15 @@ def show_data(self):
...
         payload = {"page_size": 100}
         list_of_assets = get_data(ASSET_TYPES[0], payload)
 
         # Create a new vector layer
         self.layer = create_layer(ASSET_TYPES[0], list_of_assets)
 
-        # # Close the lizard_downloader_dialog
-        # self.dlg.close()
+        # Set the status bar text
+        asset_type_caps = ASSET_TYPES[0][0].capitalize() + ASSET_TYPES[0][1:]
```
Comment (jackieleng) [inline lizard_downloader.py]: What's happening here? ``capitalize()`` already capitalizes the first letter, so shouldn't that be enough?

File: lizard_downloader.py [hunk 4]
```diff
@@ -186,18 +187,12 @@ def run(self):
                 # Create the dockwidget (after translation) and keep reference
                 self.dockwidget = Ui_DockWidget()
```
Comment (jackieleng) [inline lizard_downloader.py]: ``Ui_DockWidget()``  is also a pretty non-descriptive name

**Review by jackieleng — CHANGES_REQUESTED** [review]

File: dockwidget.py [hunk 5]
```diff
@@ -10,14 +10,14 @@
     os.path.dirname(__file__), 'dockwidget.ui'))
 
 
-class Ui_DockWidget(QtGui.QDockWidget, FORM_CLASS):
+class LizardViewer_DockWidget(QtGui.QDockWidget, FORM_CLASS):
```
Comment (jackieleng) [inline dockwidget.py]: https://www.python.org/dev/peps/pep-0008/#class-names

Underscores are not necessary.

File: lizard_downloader.py [hunk 6]
```diff
@@ -209,12 +204,17 @@ def show_data(self):
 
     def show_data(self):
         """Show the data as a new layer on the map."""
+        # Set the status bar text
+        LizardViewer_DockWidget.set_all_status_bars_text(
```
Comment (jackieleng) [inline lizard_downloader.py]: please use: `self.dockwidget.set_all_status_bars_text`

File: lizard_downloader.py [hunk 7]
```diff
@@ -209,12 +204,17 @@ def show_data(self):
...
         payload = {"page_size": 100}
         list_of_assets = get_data(ASSET_TYPES[0], payload)
 
         # Create a new vector layer
         self.layer = create_layer(ASSET_TYPES[0], list_of_assets)
 
-        # # Close the lizard_downloader_dialog
-        # self.dlg.close()
+        # Set the status bar text
+        LizardViewer_DockWidget.set_all_status_bars_text(
```
Comment (jackieleng) [inline lizard_downloader.py]: see above

File: dockwidget.py [hunk 8]
```diff
@@ -29,3 +29,15 @@ def closeEvent(self, event):
         """Module for closing the dockwidget."""
         self.closingPlugin.emit()
         event.accept()
+
+    def set_all_status_bars_text(self, string):
+        """Set the text for the status bars."""
+        status_bars = self.all_status_bars()
+        [[status_bar.setText(string), status_bar.repaint()]
+            for status_bar in status_bars]
```
Comment (jackieleng) [inline dockwidget.py]: please use a simple for loop

**Review by jackieleng — CHANGES_REQUESTED** [review]

**Pull request merged**